azure-toolkit-libs/azure-toolkit-sqlserver-lib/src/main/java/com/microsoft/azure/toolkit/lib/sqlserver/service/impl/SqlServer.java

```diff
@@ -81,9 +84,20 @@ public void delete() {
 
     @Override
     public List<SqlFirewallRuleEntity> firewallRules() {
+        if (Objects.isNull(sqlServerInner)) {
+            this.refreshInner();
+        }
         return sqlServerInner.firewallRules().list().stream().map(this::formSqlServerFirewallRule).collect(Collectors.toList());
     }
 
+    @Override
+    public List<SqlDatabaseEntity> databases() {
+        if (Objects.isNull(sqlServerInner)) {
+            this.refreshInner();
```

Comment:
wrap a help method 

 if (Objects.isNull(sqlServerInner)) {
            this.refreshInner();
        }